### PR TITLE
Changed RangeSelector to add label

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed RangeSelector to add `label`. This moves the functionality from being inside DataFilter to be part of RangeSelector directly.

#### Where should the reviewer start?

index.d.ts

#### What testing has been done on this PR?

Added a story and unit tests.

#### How should this be manually tested?

Storybook

#### Do Jest tests follow these best practices?

No change to existing unit tests for RangeSelector. They should be updated to Typescript at some point.

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### Any background context you want to provide?

Follow on from the `<Data />` work.

#### Screenshots (if appropriate)

Screenshot 2022-12-21 at 1 29 44 PM

#### Do the grommet docs need to be updated?

yes

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/RangeSelector/stories/Simple.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { RangeSelector } from '../RangeSelector';
+
+export const Simple = () => {
+  const [range, setRange] = useState();
+
+  return (
+    // Uncomment <Grommet> lines when using outside of storybook
+    // <Grommet theme={...}>
+    <Box pad="xlarge">
+      <RangeSelector
```
Comment (jcfilben) [inline src/js/components/RangeSelector/stories/Simple.js:12]: I'm no longer able to adjust the RangeSelector value in this story

**Review by jcfilben — COMMENTED** [review]
Screen Shot 2022-12-21 at 2 46 30 PM
If I have a high value the text will overflow outside of the FormField container. I think I would expect the width of the RangeSelector to reduce to accommodate for a longer text string rather than overflowing.

**Comment by ericsoderberghp** [comment]
> Screen Shot 2022-12-21 at 2 46 30 PM
> 
> If I have a high value the text will overflow outside of the FormField container. I think I would expect the width of the RangeSelector to reduce to accommodate for a longer text string rather than overflowing.

How did you set the high value? Is the upper value greater than the `max`? I'm just trying to figure out how to reproduce this locally.

File: src/js/components/RangeSelector/stories/Simple.js (hunk 1 above)
Comment (ericsoderberghp) [reply src/js/components/RangeSelector/stories/Simple.js:12]: fixed

**Comment by jcfilben** [comment]
> Screen Shot 2022-12-21 at 2 46 30 PM
> 
> If I have a high value the text will overflow outside of the FormField container. I think I would expect the width of the RangeSelector to reduce to accommodate for a longer text string rather than overflowing.

I'm still seeing this issue even after pulling down the latest changes

File: src/js/components/RangeSelector/RangeSelector.js [hunk 2]
```diff
@@ -173,6 +179,20 @@ const RangeSelector = forwardRef(
...
+    useLayoutEffect(() => {
+      if (maxRef.current && minRef.current) {
+        maxRef.current.style.width = '';
+        minRef.current.style.width = '';
+        const width = Math.max(
+          maxRef.current.getBoundingClientRect().width,
+          minRef.current.getBoundingClientRect().width,
+        );
+        maxRef.current.style.width = `${width}px`;
+        minRef.current.style.width = `${width}px`;
```
Comment (taysea) [inline src/js/components/RangeSelector/RangeSelector.js:192]: Not sure if it's related to this line, but the text is jumping side to side a little when I reach the "max" vs not:

https://user-images.githubusercontent.com/12522275/210617897-124cf17c-8b54-4a31-b91e-58c783403c5c.mov

Comment (ericsoderberghp) [reply src/js/components/RangeSelector/RangeSelector.js:194]: This should be improved with the latest changes.

**Review by taysea — APPROVED** [review]
The shift is definitely more subtle -- I am fine moving forward with the added behavior.

**Pull request merged**